integration: add tripo3d provider

dd-eg-user opened the pull request:
## Description
<!-- Provide a brief description of the changes in this PR -->
  Adds comprehensive Tripo3D provider integration to the Portkey AI Gateway, enabling users to route 3D model generation requests through Portkey to Tripo3D's API. This implementation provides full support for all Tripo3D endpoints
  including task creation, status polling, file uploads, and account management.
## Motivation
<!-- Provide a brief motivation of why the changes in this PR are needed -->
  Tripo3D is a leading 3D generation platform that converts text prompts and images into high-quality 3D models. Adding Tripo3D as a provider expands Portkey's capabilities into the 3D generation space, allowing users to leverage Portkey's
   reliability features (retries, fallbacks, load balancing) with Tripo3D's powerful 3D AI models.
## Type of Change
<!-- Put an 'x' in the boxes that apply -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)

## How Has This Been Tested?
<!-- Describe the tests you ran to verify your changes -->
- [ ] Unit Tests
- [ ] Integration Tests
- [x] Manual Testing

## Screenshots (if applicable)
<!-- Add screenshots to help explain your changes -->

## Checklist
<!-- Put an 'x' in the boxes that apply -->
- [x] My code follows the style guidelines of this project
- [ x I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Related Issues
<!-- Link related issues below. Insert the issue link or reference -->

eg-renovate-user committed:
feat: add Tripo3D provider support

Add comprehensive Tripo3D provider integration with support for:
- Task creation (text_to_model, image_to_model, etc.)
- Task status polling
- File uploads and STS token management
- Account balance checking

Implements all Tripo3D API endpoints from their OpenAPI schema.

eg-renovate-user committed:
fix: add Tripo3D endpoints to endpointStrings type

eg-renovate-user committed:
improve: replace 'any' with 'Record<string, any>' for better type safety

- Changes input field type from 'any' to 'Record<string, any>'
- Adds documentation explaining why this type is used
- Maintains flexibility while providing better type safety
- Addresses feedback about eliminating 'any' usage

eg-renovate-user committed:
add pricing estimate for tripo3d

narengogi reviewed (changes requested):
Hey @dd-eg-user 
since we do not have a unified signature for these routes,
I'd suggest only keeping the api.ts file and removing the transforms in `src/providers/tripo3d`
this would allow gateway to act as auth layer and allow passthrough proxy requests

so you index file would simply become

```js
import { ProviderConfigs } from '../types';
import Tripo3DAPIConfig from './api';

const Tripo3DConfig: ProviderConfigs = {
  api: Tripo3DAPIConfig,
};

export default Tripo3DConfig;
```

you can remove the changes in `src/providers/types.ts` as well
so finally the only required files are

`src/providers/tripo3d/api.ts`, `src/providers/tripo3d/index.ts`, `globals.ts`

dd-eg-user committed:
refactor: simplify Tripo3D provider to passthrough proxy

- Remove custom transforms and endpoint configs as requested in review
- Keep only api.ts for authentication layer functionality
- Revert endpoint string additions to types.ts
- Delete unnecessary transform files (createTask, getTask, upload, balance, pricing)
- Provider now acts as pure passthrough proxy like Sagemaker

dd-eg-user committed:
fix: update Tripo3D API to use providerPath for passthrough routing

- Change getEndpoint to use providerPath instead of hardcoded fn cases
- Fixes TypeScript errors after removing endpoint strings from types

dd-eg-user committed:
fix: use gatewayRequestURL for Tripo3D passthrough routing

- Replace providerPath with gatewayRequestURL.split('/v1')[1]
- Matches pattern used by other passthrough providers like Sagemaker
- Fixes TypeScript error in getEndpoint function

dd-eg-user committed:
remove: delete Tripo3D README for minimal passthrough provider

dd-eg-user committed:
fix: use correct passthrough pattern for Tripo3D getEndpoint

- Change from gatewayRequestURL.split('/v1')[1] to empty string return
- Matches pattern used by other simple passthrough providers like Replicate
- Addresses review comment about incorrect getEndpoint implementation
